src/selection_stream/selection_stream_processor.ts

```diff
@@ -40,7 +40,7 @@ type StatefulStream<Event, State> = {
  * Allows to select cells in the grid and update the selection
  */
 interface SelectionProcessor {
-  selectZone(anchor: AnchorZone): DispatchResult;
+  selectZone(anchor: AnchorZone, keepViewport?: boolean): DispatchResult;
```

Comment:
We try to avoid random booleans in function interface, because it makes the code harder to understand at first glance. Here I don't even think you have to do this, you can probably use `modifyAnchro` instead of `selectZone`. Look at what I did in #1988

Reply:
@hokolomopo I am not sure I get what you mean. There are also other places using `selectZone` which don't need to keep viewport (e.g. grid selection using arrow keys), and that's why I add `keepViewport` argument in `selectZone`. `modifyAnchor` is private so I cannot use it externally. Do you mean make `modifyAnchor` public?

Reply:
From outside the `SelectionProcessor` probably make `modifyAnchor` public ? I'm not even sure the SelectionProcessor should know about the concept of viewports. I don't think peoples here like API with random booleans @LucasLefevre 

Internally to the `SelectionProcessor`, my bad that's not `modifyAnchor`, but the event.type `AlterZone` which you can use that don't move the viewport. I don't think you need the `keepViewport` boolean in the event.

Reply:
> I don't think peoples here like API with random booleans @LucasLefevre

That's true. Boolean params are a code smell.
- It makes things much harder to read without knowing by heart what every argument is.
- It's also a sign the function is doing two things (violating the single responsibility principle)
- it increases the cyclomatic complexity because there must be an `if` statement somewhere. Which makes the code again less maintainable/readable

A solution is often to just have two different functions, instead of one with a param